lib/v0_bindings_extensions.rb

```diff
@@ -40,7 +40,7 @@ def list_invalid_properties
     def data=(data_object)
       *_, record_name = data_object[:type].split('.')
       swagger_class = swagger_class_for_record_name(record_name)
-      @data = swagger_class.new(data_object)
+      @data = swagger_class.new.build_from_hash(data_object)
```

Comment:
Adding `build_with_hash` uses the parsing/casting code in the autogenerated bindings, e.g. to convert `"42"` to `42`.  This makes Avro happier because it gets the types (like "integer") that it expects.